File: CodeWars-Solutions/Linked-Lists-Append.py
# Python
class Node(object):
    def __init__(self, data):
        self.data = data
        self.next = None

def append(listA, listB):
    # Your code goes here.
    # Remember to return the head of the list.
    if (listA == None and listB == None):
        return None
    if (listA == None):
        return listB
    if (listB == None):
        return listA
    
    current = listA
    while (current.next != None):
        current = current.next
    current.next = listB
    return listA

File: CodeWars-Solutions/Linked-Lists-Append.java
//Java
class Node {

  int data;
  Node next = null;

  Node(final int data) {
    this.data = data;
  }

  public Node(int data, Node next) {
    this.data = data;
    this.next = next;
  }

  public static Node append(Node listA, Node listB) {
    Node current = listA;
    if (listA == null && listB == null){
      return null;
    }
    
    if (listB == null){

    return listA;
    }
    
    if (listA == null){
      return listB;
    }
    
    while(current.next != null){
      current = current.next;
    }
    current.next = listB;
    return listA;

  }
}
